packages/react/src/Footnotes/Footnotes.tsx

```diff
@@ -0,0 +1,112 @@
+import React from 'react'
+import {ReplyIcon} from '@primer/octicons-react'
+import clsx from 'clsx'
+
+import {Heading, InlineLink, Text} from '../'
+import type {BaseProps} from '../component-helpers'
+
+/**
+ * Design tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/footnotes/colors-with-modes.css'
+
+/** * Main Stylesheet (as a CSS Module) */
+import styles from './Footnotes.module.css'
+
+export const FootnotesTags = ['div', 'ol'] as const
+
+type FootnotesRestrictedPolymorphism =
+  | (React.HTMLAttributes<HTMLDivElement> & BaseProps<HTMLDivElement> & {as?: 'div'})
+  | (React.HTMLAttributes<HTMLParagraphElement> & BaseProps<HTMLOListElement> & {as?: 'ol'})
+
+type FootnotesProps = {
+  as?: (typeof FootnotesTags)[number]
+  visuallyHiddenHeading?: string
+} & FootnotesRestrictedPolymorphism
+
+function Root({as = 'ol', className, children, visuallyHiddenHeading = 'Footnotes', ...rest}: FootnotesProps) {
```

Comment:
Just a suggestion, but do you think this component should return a `<section>`, rather than just a heading adjacent to a `div`/`ol`?

Reply:
I wondered the same but went with a fragment because these footnotes are intended to be used inside other semantically descriptive elements like `<footer>`, and not used standalone. (It will eventually be used in MinimalFooter). I think it makes the markup more verbose and complicated, but happy to add this in if there's a technical reason for needing it like improved accessibility.

Reply:
I was thinking mainly on the accessibility front, but testing with VoiceOver on Mac (admittedly it's not the greatest) it doesn't identify a `section` as a landmark. From my searching it looks like it's the same for JAWS, so happy to leave as is for now 🙂